code/clue.js

```diff
@@ -1,59 +1,201 @@
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+// AN OBJECT THAT KEEPS THE MYSTERY.
 
-// This function will be invoked when you click on the killer card.
+const mystery = {
+  killer: null,
+  weapon: null,
+  room: null
+}
+
+// This function will be invoked when you click on the KILLER CARD
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+// This will randomly select a killer from the suspects. And add that to the mystery object.
+mystery.killer = randomSelector(suspects);
+
+  document.getElementById('killerCard').style.background = mystery.killer.color;
+  document.getElementById('killerName').innerHTML = `Name: ${mystery.killer.firstName} ${mystery.killer.lastName}`;
+  document.getElementById('killerAge').innerHTML = `Age: ${mystery.killer.age}`;
+  document.getElementById('killerOccupation').innerHTML = `Occupation: ${mystery.killer.occupation}`;
+  document.getElementById('killerImage').src = mystery.killer.image;
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
 
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+// This function will be invoked when you click on the WEAPON CARD.
+  const pickWeapon = () => {
+
+  mystery.weapon = randomSelector(weapons);
+
+    document.getElementById('weaponCard').style.background = '#5F9EA0';
+    document.getElementById('weaponName').innerHTML = `${mystery.weapon.name}<br> Weight: ${mystery.weapon.weight} kg`; 
+  }
+
+  // This function will be invoked when you click on the ROOM CARD.
+  const pickRoom = () => {
+
+  mystery.room = randomSelector(rooms);
+
+    document.getElementById('roomCard').style.background = '#DCDCDC';
+    document.getElementById('roomName').innerHTML = `${mystery.room}`;  
+  }
+
+  document.getElementById('killerCard').onclick = pickKiller;
+  document.getElementById('weaponCard').onclick = pickWeapon;
+  document.getElementById('roomCard').onclick = pickRoom;
+
+// FUNCTION revealMystery that will be invoked when you click that button.
+const revealMystery = () => {
+  document.getElementById('mystery').innerHTML = `The murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName}, with a ${mystery.weapon.name} in the ${mystery.room}` 
+  document.getElementById("clickMe").onclick = revealMystery;
+}
```

Comment:
In this function, we need some sort of validation or check to make sure all the `mystery` information is available before we attempt to use it
![image](https://user-images.githubusercontent.com/807352/95061647-29569f00-06fc-11eb-9a13-c29e122260b1.png)

Reply:
I know the reveal-button does not work if you don't choose all three cards, but not sure I understand what to fix here. Can you please clarify?

Reply:
@puj Still need a bit of clarification on this 🙏